### ISSUE TITLE
⌨ `SideMenu` link labels not exposed to screen readers while closed

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#6759]
## ⌨ Accessibility Issue

Our side menu that appears on admin pages within the application has some css to hide to the labels when the menu is closed. There are two issues with this:

1. The css is hiding the labels from the accessibility tree
2. We should be able to convert this to hydrogen attributes to avoid custom css

### WCAG 2.1 Criteria

[Buttons must have discernible text](https://dequeuniversity.com/rules/axe/4.7/button-name?application=axeAPI)

## 🦋 Expected Behaviour

Even while closed, the button labels are exposed to the accessibility tree.

## 🕵️ Details

Add any additional details that could assist with troubleshooting/fixing the issue.

- **Operating System**: Name and version (if possible).
- **Browser**: Name and version (if possible).
- **Assistive Technology**: Name and version (if possible).

## 📋 Steps to Reproduce

1. Navigate to `/admin`
3. Close the side menu
4. Inpsect the accessibility tree
5. Notice the link/button labels are not present

## 📸 Screenshot
![Screenshot 2023-05-31 085732](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/9b3062c7-443b-4aee-b8be-197fc944df43)

## 🙋‍♀️ Proposed Solution

- [ ] Convert custom css to hydrogen
- [ ] Use the [`visually-hidden` hydrogen attribut](https://hydrogen.design/en/docs/properties/hydrogen/#visually-hidden)e to properly hide labels

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] Labels are visually hidden
- [ ] Labels appear in the accessibility tree
- [ ] (optional)No more custom csss
